### PR TITLE
User Story #159799830 and User Story #159800074

### PR DESCRIPTION
**Pull request by lvonah**
https://www.pivotaltracker.com/story/show/159799830 - update navbar
https://www.pivotaltracker.com/story/show/159800074 - login functionality

Team Common/Auth - Updating navbar and login functionality for interface-web

Instructions to test:
Pull code
RUN "npm install"
RUN "npm install -g @angular/cli@6.0.7" if you don't have current global angular CLI
RUN "ng serve"
Open browser to localhost:4200
Login using our credentials. Username: lvonah@project3auth.onmicrosoft.com Password: Revature!

### CODE REVIEW

File: src/app/components/navbar/navbar.component.css [hunk 1]
```diff
@@ -24,33 +24,133 @@ mat-toolbar button{
...
+.nav-toggle .icon-bar + .icon-bar {
+    margin-top: 4px;
+}
+
+.nav-dropdown{
+    align-content: center;
+    background: #484C55;
+}
+
+.nav-dropdown a{
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:hover" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-p&open=AWVJbU60iZQY_aXE08-p)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-p

File: src/app/components/navbar/navbar.component.ts [hunk 2]
```diff
@@ -1,27 +1,155 @@
...
+    }
+}
+
+/**
+ * function to set Forecast (link2) class name to active
+ * only adds active class name if it didn't have it before
+ * removes active class name from Selection (link1) class name
+ */
+
+public changeLink2() {
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.ts]: Code Smell Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 94. ([typescript:S4144](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU66iZQY_aXE08-r&open=AWVJbU66iZQY_aXE08-r)

[ISSUE_KEY]: AWVJbU66iZQY_aXE08-r

File: src/app/components/navbar/navbar.component.html [hunk 3]
```diff
@@ -1,20 +1,57 @@
...
+  <mat-grid-list cols="8" rowHeight="80px" class="nav-style">
+    <mat-grid-tile colspan="4"></mat-grid-tile>
+    <mat-grid-tile class="nav-tile" (click)="closeDropdown()">
+      <a routerLink="selection" *ngIf="LoggedIn" id="link1" class="" (click)="changeLink1()">SELECTION</a>
+    </mat-grid-tile>
+    <mat-grid-tile class="nav-tile" (click)="closeDropdown()">
+      <a routerLink="forecast" *ngIf="LoggedIn" id="link2" class="" (click)="changeLink2()">FORECAST</a>
+    </mat-grid-tile>
+    <mat-grid-tile class="nav-tile">
+      <!--  <div id="user" *ngIf="loginFailed">Username or password is incorrect</div> -->
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.html]: Code Smell Code Smell: Remove this commented out code. ([Web:AvoidCommentedOutCodeCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AAvoidCommentedOutCodeCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU6uiZQY_aXE08-n&open=AWVJbU6uiZQY_aXE08-n)

[ISSUE_KEY]: AWVJbU6uiZQY_aXE08-n

File: src/app/components/navbar/navbar.component.html [hunk 4]
```diff
@@ -1,20 +1,57 @@
...
+      </div>
+    </ul>
+  </div>
+  <div class="nav-logo" (click)="closeDropdown(); removeActive()">
+    <a routerLink="">
+      <img src="./assets/img/revature-logo.svg" alt="Revature">
+    </a>
+  </div>
+  <div class="nav-loading">
+    <img *ngIf="loading" src="../../assets/img/ajax-loader-logo.ec18efd7.gif">
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU6uiZQY_aXE08-o&open=AWVJbU6uiZQY_aXE08-o)

[ISSUE_KEY]: AWVJbU6uiZQY_aXE08-o

File: src/app/components/loading/loading.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<img src="../../assets/img/ajax-loader-logo.ec18efd7.gif">
```
Comment (fredbelotte) [inline src/app/components/loading/loading.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU7KiZQY_aXE08-t&open=AWVJbU7KiZQY_aXE08-t)

[ISSUE_KEY]: AWVJbU7KiZQY_aXE08-t

File: src/app/components/navbar/navbar.component.css (hunk 1 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:active" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-q&open=AWVJbU60iZQY_aXE08-q)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-q

File: src/app/components/navbar/navbar.component.html (hunk 3 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.html]: Code Smell Code Smell: Remove this commented out code. ([Web:AvoidCommentedOutCodeCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AAvoidCommentedOutCodeCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU6uiZQY_aXE08-n&open=AWVJbU6uiZQY_aXE08-n)

[ISSUE_KEY]: AWVJbU6uiZQY_aXE08-n

File: src/app/components/navbar/navbar.component.html (hunk 4 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU6uiZQY_aXE08-o&open=AWVJbU6uiZQY_aXE08-o)

[ISSUE_KEY]: AWVJbU6uiZQY_aXE08-o

File: src/index.html [hunk 6]
```diff
@@ -11,10 +11,12 @@
   <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
   <link href="https://3g4d13k75x47q7v53surz1gi-wpengine.netdna-ssl.com/wp-content/themes/revature/imgs/favicon.ico" rel="icon"
     type="image/x-icon">
+    
 </head>
 
 <body>
   <app-root></app-root>
+ <!-- <script type="text/javascript" src="assets/js/msal.js"></script> -->
```
Comment (fredbelotte) [inline src/index.html]: Code Smell Code Smell: Remove this commented out code. ([Web:AvoidCommentedOutCodeCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AAvoidCommentedOutCodeCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJfZkilRPS_ea-SO32&open=AWVJfZkilRPS_ea-SO32)

[ISSUE_KEY]: AWVJfZkilRPS_ea-SO32

File: src/app/components/navbar/navbar.component.css (hunk 1 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:active" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-p&open=AWVJbU60iZQY_aXE08-p)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-p

Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:hover" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-q&open=AWVJbU60iZQY_aXE08-q)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-q

File: src/app/components/loading/loading.component.html (hunk 5 above)
Comment (fredbelotte) [inline src/app/components/loading/loading.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU7KiZQY_aXE08-t&open=AWVJbU7KiZQY_aXE08-t)

[ISSUE_KEY]: AWVJbU7KiZQY_aXE08-t

File: src/app/components/navbar/navbar.component.ts (hunk 2 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.ts]: Code Smell Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 94. ([typescript:S4144](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU66iZQY_aXE08-r&open=AWVJbU66iZQY_aXE08-r)

[ISSUE_KEY]: AWVJbU66iZQY_aXE08-r

File: src/app/components/loading/loading.component.html (hunk 5 above)
Comment (fredbelotte) [inline src/app/components/loading/loading.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU7KiZQY_aXE08-t&open=AWVJbU7KiZQY_aXE08-t)

[ISSUE_KEY]: AWVJbU7KiZQY_aXE08-t

File: src/app/components/navbar/navbar.component.css (hunk 1 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:hover" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-q&open=AWVJbU60iZQY_aXE08-q)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-q

File: src/app/components/navbar/navbar.component.html [hunk 7]
```diff
@@ -1,20 +1,56 @@
...
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.html]: Bug Bug: Add an "alt" attribute to this image. ([Web:ImgWithoutAltCheck](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=Web%3AImgWithoutAltCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU6uiZQY_aXE08-o&open=AWVJbU6uiZQY_aXE08-o)

[ISSUE_KEY]: AWVJbU6uiZQY_aXE08-o

File: src/app/components/navbar/navbar.component.ts [hunk 8]
```diff
@@ -1,27 +1,137 @@
...
```
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.ts]: Code Smell Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 94. ([typescript:S4144](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU66iZQY_aXE08-r&open=AWVJbU66iZQY_aXE08-r)

[ISSUE_KEY]: AWVJbU66iZQY_aXE08-r

File: src/app/components/navbar/navbar.component.css (hunk 1 above)
Comment (fredbelotte) [inline src/app/components/navbar/navbar.component.css]: Code Smell Code Smell: Expected selector ".nav-dropdown a" to come before selector ".nav-tile a:active" ([css:S4664](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=css%3AS4664))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=interface%3Aweb&pullRequest=24&issues=AWVJbU60iZQY_aXE08-p&open=AWVJbU60iZQY_aXE08-p)

[ISSUE_KEY]: AWVJbU60iZQY_aXE08-p

File: dockerup.yml [hunk 9]
```diff
@@ -5,7 +5,7 @@ networks:
 
 services:
   web:
-    image: revaturecloud/interface-web:dev
+    image: interface-web:latest
```
Comment (escalonn) [inline dockerup.yml]: this file is used for the continuous deployment. make this other version in a separate yml file following the convention of selection team.

File: src/app/app.component.html [hunk 10]
```diff
@@ -1,4 +1,7 @@
 <!--The content below is only a placeholder and can be replaced.-->
+
 <app-navbar></app-navbar>
 <router-outlet></router-outlet>
 <app-footer></app-footer>
+
+
```
Comment (escalonn) [inline src/app/app.component.html]: unnecessary whitespace added (we could stand to delete that "placeholder" comment though.)

File: src/app/app.component.ts [hunk 11]
```diff
@@ -1,5 +1,6 @@
 import { Component } from '@angular/core';
 
+
```
Comment (escalonn) [inline src/app/app.component.ts]: no need for this

File: src/app/components/navbar/navbar.component.html [hunk 12]
```diff
@@ -1,20 +1,56 @@
...
+    </ul>
+  </div>
+  <div class="nav-logo" (click)="closeDropdown(); removeActive()">
+    <a routerLink="">
+      <img src="./assets/img/revature-logo.svg" alt="Revature">
+    </a>
+  </div>
+  <div class="nav-loading">
+    <img *ngIf="loading" src="../../assets/img/ajax-loader-logo.ec18efd7.gif" alt="Loading">
+  </div>
```
Comment (escalonn) [inline src/app/components/navbar/navbar.component.html]: either delete the separate loading component, or use it here, no sense having both

File: src/app/components/navbar/navbar.component.ts [hunk 13]
```diff
@@ -1,27 +1,124 @@
...
+  constructor(private authService: AuthService, private router: Router) { }
+  
 
   /**
    * Lifecycle hook that is called after data-bound properties of a directive are initialized.
    * https://angular.io/api/core/OnInit
    */
   ngOnInit() {
+      if (this.authService.isLoggedIn)
+      {
```
Comment (escalonn) [inline src/app/components/navbar/navbar.component.ts]: we put the brace on the same line in JS/TS

File: src/app/components/navbar/navbar.component.ts [hunk 14]
```diff
@@ -1,27 +1,124 @@
...
+      switch (window.localStorage.getItem("activeTile")) {
+          case "0":
+              break;
+      
+          case "link1":
+              document.getElementById("link1").classList.add("active");
+              break;
+
+          case "link2":
+              document.getElementById("link2").classList.add("active");
```
Comment (escalonn) [inline src/app/components/navbar/navbar.component.ts]: use template reference variables and ViewChild, not ids and getElementById

File: src/app/services/auth.service.ts [hunk 15]
```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+//import './node_modules/msal/out/msal';
```
Comment (escalonn) [inline src/app/services/auth.service.ts]: remove commented out code

File: src/index.html [hunk 16]
```diff
@@ -11,6 +11,7 @@
   <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
   <link href="https://3g4d13k75x47q7v53surz1gi-wpengine.netdna-ssl.com/wp-content/themes/revature/imgs/favicon.ico" rel="icon"
     type="image/x-icon">
+
```
Comment (escalonn) [inline src/index.html]: unneeded

File: src/app/animations.ts [hunk 17]
```diff
@@ -0,0 +1,38 @@
+import { trigger, state, style, transition,
+    animate, group, query, stagger, keyframes
+} from '@angular/animations';
```
Comment (escalonn) [inline src/app/animations.ts]: this doesn't look like great style

File: src/app/components/navbar/navbar.component.ts [hunk 18]
```diff
@@ -1,27 +1,77 @@
...
+    public firstName: string[];
+    public animationState = 'in';
+
+    constructor(private authService: AuthService, private router: Router, private activatedRoute: ActivatedRoute) { }
+
+
+    /**
+     * Lifecycle hook that is called after data-bound properties of a directive are initialized.
+     * https://angular.io/api/core/OnInit
+     */
```
Comment (escalonn) [inline src/app/components/navbar/navbar.component.ts]: this comment is pretty superfluous, this is an angular basic

File: src/app/guards/auth.guard.ts [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+ * they will be redirected to '' (Homepage)
+ */
+
+export class AuthGuard implements CanActivate {
+  // add the service we need
+  constructor(
+    private auth: AuthService,
+    private router: Router
+  ) {}
+  canActivate(
```
Comment (escalonn) [inline src/app/guards/auth.guard.ts]: should line break between methods
